lib/whitehall_importer/create_revision.rb

```diff
@@ -82,30 +82,34 @@ def translation
       @translation || raise(AbortImportError, "Translation #{document.locale} missing")
     end
 
-    def primary_publishing_organisation(organisations)
-      unless organisations
+    def primary_publishing_organisation
+      unless whitehall_edition["organisations"]
         raise AbortImportError, "Must have at least one organisation"
       end
 
-      primary_publishing_organisations = organisations.select do |organisation|
-        organisation["lead"]
-      end
-
       unless primary_publishing_organisations.any?
         raise AbortImportError, "Lead organisation missing"
       end
 
-      primary_publishing_organisation = primary_publishing_organisations.min { |o| o["lead_ordering"] }
+      [first_lead_organisation["content_id"]]
+    end
 
-      [primary_publishing_organisation["content_id"]]
+    def supporting_organisations
+      supporting_organisations = whitehall_edition["organisations"].reject do |organisation|
+        first_lead_organisation == organisation
+      end
+
+      supporting_organisations.map { |organisation| organisation["content_id"] }
     end
 
-    def supporting_organisations(organisations)
-      supporting_organisations = organisations.reject do |organisation|
+    def primary_publishing_organisations
+      whitehall_edition["organisations"].select do |organisation|
         organisation["lead"]
       end
+    end
 
-      supporting_organisations.map { |organisation| organisation["content_id"] }
+    def first_lead_organisation
+      @first_lead_organisation ||= primary_publishing_organisations.min { |o| o["lead_ordering"] }
```

Comment:
I'm wondering if it might be an idea to sort the return value from `primary_publishing_organisations`, then this method probably wouldn't be needed (since you could just use `.first` in `primary_publishing_organisation`)?

Reply:
Sure, you do lose the memoization aspect but as we're only looping through hash's on small arrays it doesn't really make a difference!